test: TestLeaseKeepAliveCloseAfterDisconnectRevoke unexpected keepalive

via semaphore

```
=== RUN   TestLeaseKeepAliveCloseAfterDisconnectRevoke
2016-07-08 00:27:52.070712 I | integration: launching node5021162303408691792 (unix://localhost:node5021162303408691792.sock.bridge)
2016-07-08 00:27:52.071727 I | integration: launching node2587048822844350318 (unix://localhost:node2587048822844350318.sock.bridge)
2016-07-08 00:27:52.072317 I | integration: launching node6227899913783768578 (unix://localhost:node6227899913783768578.sock.bridge)
2016-07-08 00:27:52.091658 I | etcdserver: name = node5021162303408691792
2016-07-08 00:27:52.091708 I | etcdserver: data dir = /tmp/etcd553609219
2016-07-08 00:27:52.091737 I | etcdserver: member dir = /tmp/etcd553609219/member
2016-07-08 00:27:52.091768 I | etcdserver: heartbeat = 10ms
2016-07-08 00:27:52.091801 I | etcdserver: election = 100ms
2016-07-08 00:27:52.091832 I | etcdserver: snapshot count = 0
2016-07-08 00:27:52.091876 I | etcdserver: advertise client URLs = unix://127.0.0.1:21122.29554.sock
2016-07-08 00:27:52.091915 I | etcdserver: initial advertise peer URLs = unix://127.0.0.1:21121.29554.sock
2016-07-08 00:27:52.092069 I | etcdserver: initial cluster = node2587048822844350318=unix://127.0.0.1:21119.29554.sock,node5021162303408691792=unix://127.0.0.1:21121.29554.sock,node6227899913783768578=unix://127.0.0.1:21123.29554.sock
2016-07-08 00:27:52.092605 I | etcdserver: name = node2587048822844350318
2016-07-08 00:27:52.092648 I | etcdserver: data dir = /tmp/etcd192011828
2016-07-08 00:27:52.092684 I | etcdserver: member dir = /tmp/etcd192011828/member
2016-07-08 00:27:52.092709 I | etcdserver: heartbeat = 10ms
2016-07-08 00:27:52.092733 I | etcdserver: election = 100ms
2016-07-08 00:27:52.092759 I | etcdserver: snapshot count = 0
2016-07-08 00:27:52.092790 I | etcdserver: advertise client URLs = unix://127.0.0.1:21120.29554.sock
2016-07-08 00:27:52.092823 I | etcdserver: initial advertise peer URLs = unix://127.0.0.1:21119.29554.sock
2016-07-08 00:27:52.092929 I | etcdserver: initial cluster = node2587048822844350318=unix://127.0.0.1:21119.29554.sock,node5021162303408691792=unix://127.0.0.1:21121.29554.sock,node6227899913783768578=unix://127.0.0.1:21123.29554.sock
2016-07-08 00:27:52.093472 I | etcdserver: name = node6227899913783768578
2016-07-08 00:27:52.093533 I | etcdserver: data dir = /tmp/etcd949902726
2016-07-08 00:27:52.093587 I | etcdserver: member dir = /tmp/etcd949902726/member
2016-07-08 00:27:52.093634 I | etcdserver: heartbeat = 10ms
2016-07-08 00:27:52.093677 I | etcdserver: election = 100ms
2016-07-08 00:27:52.093719 I | etcdserver: snapshot count = 0
2016-07-08 00:27:52.093770 I | etcdserver: advertise client URLs = unix://127.0.0.1:21124.29554.sock
2016-07-08 00:27:52.093823 I | etcdserver: initial advertise peer URLs = unix://127.0.0.1:21123.29554.sock
2016-07-08 00:27:52.093981 I | etcdserver: initial cluster = node2587048822844350318=unix://127.0.0.1:21119.29554.sock,node5021162303408691792=unix://127.0.0.1:21121.29554.sock,node6227899913783768578=unix://127.0.0.1:21123.29554.sock
2016-07-08 00:27:52.102052 I | etcdserver: starting member a3da7b6cbdc85c13 in cluster 14b0d63db98d5990
2016-07-08 00:27:52.102169 I | raft: a3da7b6cbdc85c13 became follower at term 0
2016-07-08 00:27:52.102225 I | raft: newRaft a3da7b6cbdc85c13 [peers: [], term: 0, commit: 0, applied: 0, lastindex: 0, lastterm: 0]
2016-07-08 00:27:52.102261 I | raft: a3da7b6cbdc85c13 became follower at term 1
2016-07-08 00:27:52.103877 I | rafthttp: starting peer 3299bc18a0c7b5b0...
2016-07-08 00:27:52.104167 I | rafthttp: started HTTP pipelining with peer 3299bc18a0c7b5b0
2016-07-08 00:27:52.108970 I | etcdserver: starting member 3299bc18a0c7b5b0 in cluster 14b0d63db98d5990
2016-07-08 00:27:52.109127 I | raft: 3299bc18a0c7b5b0 became follower at term 0
2016-07-08 00:27:52.109209 I | raft: newRaft 3299bc18a0c7b5b0 [peers: [], term: 0, commit: 0, applied: 0, lastindex: 0, lastterm: 0]
2016-07-08 00:27:52.109261 I | raft: 3299bc18a0c7b5b0 became follower at term 1
2016-07-08 00:27:52.111293 I | rafthttp: starting peer a3da7b6cbdc85c13...
2016-07-08 00:27:52.129679 I | etcdserver: starting member d6ac5313ae37eb88 in cluster 14b0d63db98d5990
2016-07-08 00:27:52.136556 I | raft: d6ac5313ae37eb88 became follower at term 0
2016-07-08 00:27:52.136605 I | raft: newRaft d6ac5313ae37eb88 [peers: [], term: 0, commit: 0, applied: 0, lastindex: 0, lastterm: 0]
2016-07-08 00:27:52.136635 I | raft: d6ac5313ae37eb88 became follower at term 1
2016-07-08 00:27:52.138002 I | rafthttp: starting peer 3299bc18a0c7b5b0...
2016-07-08 00:27:52.138259 I | rafthttp: started HTTP pipelining with peer 3299bc18a0c7b5b0
2016-07-08 00:27:52.141514 I | rafthttp: started peer 3299bc18a0c7b5b0
2016-07-08 00:27:52.141729 I | rafthttp: added peer 3299bc18a0c7b5b0
2016-07-08 00:27:52.141774 I | rafthttp: starting peer a3da7b6cbdc85c13...
2016-07-08 00:27:52.142002 I | rafthttp: started HTTP pipelining with peer a3da7b6cbdc85c13
2016-07-08 00:27:52.146103 I | rafthttp: started peer a3da7b6cbdc85c13
2016-07-08 00:27:52.146304 I | rafthttp: added peer a3da7b6cbdc85c13
2016-07-08 00:27:52.146347 I | etcdserver: set snapshot count to default 10000
2016-07-08 00:27:52.146395 I | etcdserver: starting server... [version: 3.0.0+git, cluster version: to_be_decided]
2016-07-08 00:27:52.146874 I | rafthttp: started peer 3299bc18a0c7b5b0
2016-07-08 00:27:52.147042 I | rafthttp: added peer 3299bc18a0c7b5b0
2016-07-08 00:27:52.147110 I | rafthttp: starting peer d6ac5313ae37eb88...
2016-07-08 00:27:52.147390 I | rafthttp: started HTTP pipelining with peer d6ac5313ae37eb88
2016-07-08 00:27:52.150454 I | rafthttp: started peer d6ac5313ae37eb88
2016-07-08 00:27:52.150614 I | rafthttp: added peer d6ac5313ae37eb88
2016-07-08 00:27:52.150648 I | etcdserver: set snapshot count to default 10000
2016-07-08 00:27:52.150677 I | etcdserver: starting server... [version: 3.0.0+git, cluster version: to_be_decided]
2016-07-08 00:27:52.151791 I | rafthttp: started streaming with peer 3299bc18a0c7b5b0 (writer)
2016-07-08 00:27:52.151872 I | rafthttp: started streaming with peer 3299bc18a0c7b5b0 (writer)
2016-07-08 00:27:52.151936 I | rafthttp: started streaming with peer 3299bc18a0c7b5b0 (stream MsgApp v2 reader)
2016-07-08 00:27:52.152142 I | rafthttp: started streaming with peer 3299bc18a0c7b5b0 (stream Message reader)
2016-07-08 00:27:52.152352 I | rafthttp: started streaming with peer a3da7b6cbdc85c13 (writer)
2016-07-08 00:27:52.152419 I | rafthttp: started streaming with peer a3da7b6cbdc85c13 (writer)
2016-07-08 00:27:52.152489 I | rafthttp: started streaming with peer a3da7b6cbdc85c13 (stream MsgApp v2 reader)
2016-07-08 00:27:52.152864 I | rafthttp: started streaming with peer a3da7b6cbdc85c13 (stream Message reader)
2016-07-08 00:27:52.153711 I | rafthttp: started streaming with peer 3299bc18a0c7b5b0 (writer)
2016-07-08 00:27:52.153846 I | rafthttp: started streaming with peer 3299bc18a0c7b5b0 (writer)
2016-07-08 00:27:52.153972 I | rafthttp: started streaming with peer d6ac5313ae37eb88 (writer)
2016-07-08 00:27:52.154029 I | rafthttp: started streaming with peer d6ac5313ae37eb88 (writer)
2016-07-08 00:27:52.154092 I | rafthttp: started streaming with peer d6ac5313ae37eb88 (stream MsgApp v2 reader)
2016-07-08 00:27:52.154300 I | rafthttp: started streaming with peer d6ac5313ae37eb88 (stream Message reader)
2016-07-08 00:27:52.156028 I | rafthttp: started streaming with peer 3299bc18a0c7b5b0 (stream MsgApp v2 reader)
2016-07-08 00:27:52.157773 I | integration: launched node5021162303408691792 (unix://localhost:node5021162303408691792.sock.bridge)
2016-07-08 00:27:52.158249 I | rafthttp: started streaming with peer 3299bc18a0c7b5b0 (stream Message reader)
2016-07-08 00:27:52.159408 I | rafthttp: peer d6ac5313ae37eb88 became active
2016-07-08 00:27:52.159455 I | rafthttp: established a TCP streaming connection with peer d6ac5313ae37eb88 (stream MsgApp v2 writer)
2016-07-08 00:27:52.160839 I | membership: added member 3299bc18a0c7b5b0 [unix://127.0.0.1:21119.29554.sock] to cluster 14b0d63db98d5990
2016-07-08 00:27:52.161069 I | rafthttp: started HTTP pipelining with peer a3da7b6cbdc85c13
2016-07-08 00:27:52.164036 I | rafthttp: started peer a3da7b6cbdc85c13
2016-07-08 00:27:52.164222 I | rafthttp: added peer a3da7b6cbdc85c13
2016-07-08 00:27:52.164285 I | rafthttp: starting peer d6ac5313ae37eb88...
2016-07-08 00:27:52.164541 I | rafthttp: started HTTP pipelining with peer d6ac5313ae37eb88
2016-07-08 00:27:52.189013 I | rafthttp: started streaming with peer d6ac5313ae37eb88 (writer)
2016-07-08 00:27:52.189160 I | raft: a3da7b6cbdc85c13 is starting a new election at term 1
2016-07-08 00:27:52.189251 I | raft: a3da7b6cbdc85c13 became candidate at term 2
2016-07-08 00:27:52.189302 I | raft: a3da7b6cbdc85c13 received vote from a3da7b6cbdc85c13 at term 2
2016-07-08 00:27:52.189369 I | raft: a3da7b6cbdc85c13 [logterm: 1, index: 3] sent vote request to 3299bc18a0c7b5b0 at term 2
2016-07-08 00:27:52.189438 I | raft: a3da7b6cbdc85c13 [logterm: 1, index: 3] sent vote request to d6ac5313ae37eb88 at term 2
2016-07-08 00:27:52.192991 I | rafthttp: started peer d6ac5313ae37eb88
2016-07-08 00:27:52.193187 I | rafthttp: added peer d6ac5313ae37eb88
2016-07-08 00:27:52.193239 I | etcdserver: set snapshot count to default 10000
2016-07-08 00:27:52.193283 I | etcdserver: starting server... [version: 3.0.0+git, cluster version: to_be_decided]
2016-07-08 00:27:52.195175 I | membership: added member 3299bc18a0c7b5b0 [unix://127.0.0.1:21119.29554.sock] to cluster 14b0d63db98d5990
2016-07-08 00:27:52.196618 I | rafthttp: peer a3da7b6cbdc85c13 became active
2016-07-08 00:27:52.196683 I | rafthttp: established a TCP streaming connection with peer a3da7b6cbdc85c13 (stream MsgApp v2 writer)
2016-07-08 00:27:52.197096 I | rafthttp: started streaming with peer a3da7b6cbdc85c13 (writer)
2016-07-08 00:27:52.197176 I | rafthttp: started streaming with peer a3da7b6cbdc85c13 (writer)
2016-07-08 00:27:52.197260 I | rafthttp: started streaming with peer a3da7b6cbdc85c13 (stream MsgApp v2 reader)
2016-07-08 00:27:52.197750 I | rafthttp: started streaming with peer a3da7b6cbdc85c13 (stream Message reader)
2016-07-08 00:27:52.198333 I | rafthttp: established a TCP streaming connection with peer a3da7b6cbdc85c13 (stream MsgApp v2 reader)
2016-07-08 00:27:52.199476 I | membership: added member a3da7b6cbdc85c13 [unix://127.0.0.1:21121.29554.sock] to cluster 14b0d63db98d5990
2016-07-08 00:27:52.199988 I | rafthttp: started streaming with peer d6ac5313ae37eb88 (writer)
2016-07-08 00:27:52.200151 I | rafthttp: peer d6ac5313ae37eb88 became active
2016-07-08 00:27:52.200188 I | rafthttp: established a TCP streaming connection with peer d6ac5313ae37eb88 (stream Message writer)
2016-07-08 00:27:52.200369 I | rafthttp: started streaming with peer d6ac5313ae37eb88 (stream MsgApp v2 reader)
2016-07-08 00:27:52.200771 I | rafthttp: started streaming with peer d6ac5313ae37eb88 (stream Message reader)
2016-07-08 00:27:52.202967 I | rafthttp: established a TCP streaming connection with peer d6ac5313ae37eb88 (stream MsgApp v2 writer)
2016-07-08 00:27:52.203564 I | membership: added member a3da7b6cbdc85c13 [unix://127.0.0.1:21121.29554.sock] to cluster 14b0d63db98d5990
2016-07-08 00:27:52.204768 I | rafthttp: established a TCP streaming connection with peer d6ac5313ae37eb88 (stream Message writer)
2016-07-08 00:27:52.204879 I | raft: d6ac5313ae37eb88 is starting a new election at term 1
2016-07-08 00:27:52.204947 I | raft: d6ac5313ae37eb88 became candidate at term 2
2016-07-08 00:27:52.204991 I | raft: d6ac5313ae37eb88 received vote from d6ac5313ae37eb88 at term 2
2016-07-08 00:27:52.205046 I | raft: d6ac5313ae37eb88 [logterm: 1, index: 3] sent vote request to a3da7b6cbdc85c13 at term 2
2016-07-08 00:27:52.205101 I | raft: d6ac5313ae37eb88 [logterm: 1, index: 3] sent vote request to 3299bc18a0c7b5b0 at term 2
2016-07-08 00:27:52.206807 I | membership: added member 3299bc18a0c7b5b0 [unix://127.0.0.1:21119.29554.sock] to cluster 14b0d63db98d5990
2016-07-08 00:27:52.207551 I | rafthttp: peer a3da7b6cbdc85c13 became active
2016-07-08 00:27:52.207609 I | rafthttp: established a TCP streaming connection with peer a3da7b6cbdc85c13 (stream MsgApp v2 writer)
2016-07-08 00:27:52.207868 I | membership: added member d6ac5313ae37eb88 [unix://127.0.0.1:21123.29554.sock] to cluster 14b0d63db98d5990
2016-07-08 00:27:52.207974 I | rafthttp: established a TCP streaming connection with peer a3da7b6cbdc85c13 (stream Message writer)
2016-07-08 00:27:52.208325 I | membership: added member d6ac5313ae37eb88 [unix://127.0.0.1:21123.29554.sock] to cluster 14b0d63db98d5990
2016-07-08 00:27:52.208401 W | etcdserver: apply entries took too long [50.312245ms for 3 entries]
2016-07-08 00:27:52.208435 W | etcdserver: avoid queries with large range/delete range!
2016-07-08 00:27:52.209579 I | membership: added member a3da7b6cbdc85c13 [unix://127.0.0.1:21121.29554.sock] to cluster 14b0d63db98d5990
2016-07-08 00:27:52.209986 I | membership: added member d6ac5313ae37eb88 [unix://127.0.0.1:21123.29554.sock] to cluster 14b0d63db98d5990
2016-07-08 00:27:52.210633 I | integration: launched node6227899913783768578 (unix://localhost:node6227899913783768578.sock.bridge)
2016-07-08 00:27:52.211018 I | rafthttp: established a TCP streaming connection with peer d6ac5313ae37eb88 (stream MsgApp v2 reader)
2016-07-08 00:27:52.211603 I | raft: 3299bc18a0c7b5b0 is starting a new election at term 1
2016-07-08 00:27:52.211665 I | raft: 3299bc18a0c7b5b0 became candidate at term 2
2016-07-08 00:27:52.211706 I | raft: 3299bc18a0c7b5b0 received vote from 3299bc18a0c7b5b0 at term 2
2016-07-08 00:27:52.211756 I | raft: 3299bc18a0c7b5b0 [logterm: 1, index: 3] sent vote request to d6ac5313ae37eb88 at term 2
2016-07-08 00:27:52.211806 I | raft: 3299bc18a0c7b5b0 [logterm: 1, index: 3] sent vote request to a3da7b6cbdc85c13 at term 2
2016-07-08 00:27:52.213959 I | integration: launched node2587048822844350318 (unix://localhost:node2587048822844350318.sock.bridge)
2016-07-08 00:27:52.215155 I | rafthttp: peer 3299bc18a0c7b5b0 became active
2016-07-08 00:27:52.215200 I | rafthttp: established a TCP streaming connection with peer 3299bc18a0c7b5b0 (stream Message reader)
2016-07-08 00:27:52.215919 I | raft: 3299bc18a0c7b5b0 [logterm: 1, index: 3, vote: 3299bc18a0c7b5b0] rejected vote from d6ac5313ae37eb88 [logterm: 1, index: 3] at term 2
2016-07-08 00:27:52.216860 I | rafthttp: established a TCP streaming connection with peer a3da7b6cbdc85c13 (stream Message reader)
2016-07-08 00:27:52.217383 I | raft: d6ac5313ae37eb88 [logterm: 1, index: 3, vote: d6ac5313ae37eb88] rejected vote from 3299bc18a0c7b5b0 [logterm: 1, index: 3] at term 2
2016-07-08 00:27:52.217436 I | raft: d6ac5313ae37eb88 [logterm: 1, index: 3, vote: d6ac5313ae37eb88] rejected vote from a3da7b6cbdc85c13 [logterm: 1, index: 3] at term 2
2016-07-08 00:27:52.217943 I | raft: 3299bc18a0c7b5b0 [logterm: 1, index: 3, vote: 3299bc18a0c7b5b0] rejected vote from a3da7b6cbdc85c13 [logterm: 1, index: 3] at term 2
2016-07-08 00:27:52.218026 I | rafthttp: established a TCP streaming connection with peer d6ac5313ae37eb88 (stream Message reader)
2016-07-08 00:27:52.218095 I | rafthttp: established a TCP streaming connection with peer a3da7b6cbdc85c13 (stream Message writer)
2016-07-08 00:27:52.218153 I | rafthttp: peer 3299bc18a0c7b5b0 became active
2016-07-08 00:27:52.218192 I | rafthttp: established a TCP streaming connection with peer 3299bc18a0c7b5b0 (stream MsgApp v2 reader)
2016-07-08 00:27:52.218723 I | rafthttp: established a TCP streaming connection with peer 3299bc18a0c7b5b0 (stream MsgApp v2 reader)
2016-07-08 00:27:52.219616 I | rafthttp: established a TCP streaming connection with peer 3299bc18a0c7b5b0 (stream Message writer)
2016-07-08 00:27:52.220085 I | rafthttp: established a TCP streaming connection with peer 3299bc18a0c7b5b0 (stream Message writer)
2016-07-08 00:27:52.221049 I | rafthttp: established a TCP streaming connection with peer 3299bc18a0c7b5b0 (stream MsgApp v2 writer)
2016-07-08 00:27:52.221206 I | raft: a3da7b6cbdc85c13 [logterm: 1, index: 3, vote: a3da7b6cbdc85c13] rejected vote from d6ac5313ae37eb88 [logterm: 1, index: 3] at term 2
2016-07-08 00:27:52.221278 I | raft: a3da7b6cbdc85c13 [logterm: 1, index: 3, vote: a3da7b6cbdc85c13] rejected vote from 3299bc18a0c7b5b0 [logterm: 1, index: 3] at term 2
2016-07-08 00:27:52.222160 I | rafthttp: established a TCP streaming connection with peer 3299bc18a0c7b5b0 (stream MsgApp v2 writer)
2016-07-08 00:27:52.223159 I | rafthttp: established a TCP streaming connection with peer 3299bc18a0c7b5b0 (stream Message reader)
2016-07-08 00:27:52.223307 I | raft: a3da7b6cbdc85c13 received vote rejection from 3299bc18a0c7b5b0 at term 2
2016-07-08 00:27:52.223356 I | raft: a3da7b6cbdc85c13 [quorum:2] has received 1 votes and 1 vote rejections
2016-07-08 00:27:52.224122 I | raft: d6ac5313ae37eb88 received vote rejection from 3299bc18a0c7b5b0 at term 2
2016-07-08 00:27:52.224162 I | raft: d6ac5313ae37eb88 [quorum:2] has received 1 votes and 1 vote rejections
2016-07-08 00:27:52.225412 I | rafthttp: established a TCP streaming connection with peer d6ac5313ae37eb88 (stream MsgApp v2 reader)
2016-07-08 00:27:52.226233 I | rafthttp: established a TCP streaming connection with peer a3da7b6cbdc85c13 (stream MsgApp v2 reader)
2016-07-08 00:27:52.227034 I | raft: a3da7b6cbdc85c13 received vote rejection from d6ac5313ae37eb88 at term 2
2016-07-08 00:27:52.227081 I | raft: a3da7b6cbdc85c13 [quorum:2] has received 1 votes and 2 vote rejections
2016-07-08 00:27:52.227136 I | raft: a3da7b6cbdc85c13 became follower at term 2
2016-07-08 00:27:52.227235 I | raft: d6ac5313ae37eb88 received vote rejection from a3da7b6cbdc85c13 at term 2
2016-07-08 00:27:52.227273 I | raft: d6ac5313ae37eb88 [quorum:2] has received 1 votes and 2 vote rejections
2016-07-08 00:27:52.227316 I | raft: d6ac5313ae37eb88 became follower at term 2
2016-07-08 00:27:52.227444 I | rafthttp: established a TCP streaming connection with peer a3da7b6cbdc85c13 (stream Message reader)
2016-07-08 00:27:52.227584 I | rafthttp: established a TCP streaming connection with peer d6ac5313ae37eb88 (stream Message reader)
2016-07-08 00:27:52.227741 I | raft: 3299bc18a0c7b5b0 received vote rejection from a3da7b6cbdc85c13 at term 2
2016-07-08 00:27:52.227775 I | raft: 3299bc18a0c7b5b0 [quorum:2] has received 1 votes and 1 vote rejections
2016-07-08 00:27:52.227815 I | raft: 3299bc18a0c7b5b0 received vote rejection from d6ac5313ae37eb88 at term 2
2016-07-08 00:27:52.227843 I | raft: 3299bc18a0c7b5b0 [quorum:2] has received 1 votes and 2 vote rejections
2016-07-08 00:27:52.227883 I | raft: 3299bc18a0c7b5b0 became follower at term 2
2016-07-08 00:27:52.394300 I | raft: a3da7b6cbdc85c13 is starting a new election at term 2
2016-07-08 00:27:52.394671 I | raft: a3da7b6cbdc85c13 became candidate at term 3
2016-07-08 00:27:52.394875 I | raft: a3da7b6cbdc85c13 received vote from a3da7b6cbdc85c13 at term 3
2016-07-08 00:27:52.395148 I | raft: a3da7b6cbdc85c13 [logterm: 1, index: 3] sent vote request to 3299bc18a0c7b5b0 at term 3
2016-07-08 00:27:52.395434 I | raft: a3da7b6cbdc85c13 [logterm: 1, index: 3] sent vote request to d6ac5313ae37eb88 at term 3
2016-07-08 00:27:52.397076 I | raft: d6ac5313ae37eb88 is starting a new election at term 2
2016-07-08 00:27:52.397323 I | raft: d6ac5313ae37eb88 became candidate at term 3
2016-07-08 00:27:52.397508 I | raft: d6ac5313ae37eb88 received vote from d6ac5313ae37eb88 at term 3
2016-07-08 00:27:52.397676 I | raft: d6ac5313ae37eb88 [logterm: 1, index: 3] sent vote request to 3299bc18a0c7b5b0 at term 3
2016-07-08 00:27:52.397836 I | raft: d6ac5313ae37eb88 [logterm: 1, index: 3] sent vote request to a3da7b6cbdc85c13 at term 3
2016-07-08 00:27:52.399350 I | raft: 3299bc18a0c7b5b0 [term: 2] received a MsgVote message with higher term from d6ac5313ae37eb88 [term: 3]
2016-07-08 00:27:52.399587 I | raft: 3299bc18a0c7b5b0 became follower at term 3
2016-07-08 00:27:52.399782 I | raft: 3299bc18a0c7b5b0 [logterm: 1, index: 3, vote: 0] voted for d6ac5313ae37eb88 [logterm: 1, index: 3] at term 3
2016-07-08 00:27:52.401617 I | raft: 3299bc18a0c7b5b0 [logterm: 1, index: 3, vote: d6ac5313ae37eb88] rejected vote from a3da7b6cbdc85c13 [logterm: 1, index: 3] at term 3
2016-07-08 00:27:52.401932 I | raft: d6ac5313ae37eb88 [logterm: 1, index: 3, vote: d6ac5313ae37eb88] rejected vote from a3da7b6cbdc85c13 [logterm: 1, index: 3] at term 3
2016-07-08 00:27:52.402308 I | raft: a3da7b6cbdc85c13 [logterm: 1, index: 3, vote: a3da7b6cbdc85c13] rejected vote from d6ac5313ae37eb88 [logterm: 1, index: 3] at term 3
2016-07-08 00:27:52.404322 I | raft: d6ac5313ae37eb88 received vote rejection from a3da7b6cbdc85c13 at term 3
2016-07-08 00:27:52.404532 I | raft: d6ac5313ae37eb88 [quorum:2] has received 1 votes and 1 vote rejections
2016-07-08 00:27:52.405718 I | raft: a3da7b6cbdc85c13 received vote rejection from 3299bc18a0c7b5b0 at term 3
2016-07-08 00:27:52.405886 I | raft: a3da7b6cbdc85c13 [quorum:2] has received 1 votes and 1 vote rejections
2016-07-08 00:27:52.406089 I | raft: a3da7b6cbdc85c13 received vote rejection from d6ac5313ae37eb88 at term 3
2016-07-08 00:27:52.406211 I | raft: a3da7b6cbdc85c13 [quorum:2] has received 1 votes and 2 vote rejections
2016-07-08 00:27:52.406362 I | raft: a3da7b6cbdc85c13 became follower at term 3
2016-07-08 00:27:52.406558 I | raft: d6ac5313ae37eb88 received vote from 3299bc18a0c7b5b0 at term 3
2016-07-08 00:27:52.406679 I | raft: d6ac5313ae37eb88 [quorum:2] has received 2 votes and 1 vote rejections
2016-07-08 00:27:52.406846 I | raft: d6ac5313ae37eb88 became leader at term 3
2016-07-08 00:27:52.407044 I | raft: raft.node: d6ac5313ae37eb88 elected leader d6ac5313ae37eb88 at term 3
2016-07-08 00:27:52.410540 I | raft: raft.node: a3da7b6cbdc85c13 elected leader d6ac5313ae37eb88 at term 3
2016-07-08 00:27:52.413551 I | raft: raft.node: 3299bc18a0c7b5b0 elected leader d6ac5313ae37eb88 at term 3
2016-07-08 00:27:52.447328 I | etcdserver: published {Name:node6227899913783768578 ClientURLs:[unix://127.0.0.1:21124.29554.sock]} to cluster 14b0d63db98d5990
2016-07-08 00:27:52.456367 I | etcdserver: published {Name:node5021162303408691792 ClientURLs:[unix://127.0.0.1:21122.29554.sock]} to cluster 14b0d63db98d5990
2016-07-08 00:27:52.465284 I | etcdserver: setting up the initial cluster version to 3.0
2016-07-08 00:27:52.467145 I | etcdserver: published {Name:node2587048822844350318 ClientURLs:[unix://127.0.0.1:21120.29554.sock]} to cluster 14b0d63db98d5990
2016-07-08 00:27:52.482234 N | membership: set the initial cluster version to 3.0
2016-07-08 00:27:52.488942 N | membership: set the initial cluster version to 3.0
2016-07-08 00:27:52.490156 N | membership: set the initial cluster version to 3.0
2016-07-08 00:27:52.576720 I | integration: stopping node2587048822844350318 (unix://localhost:node2587048822844350318.sock.bridge)
2016-07-08 00:27:52.579171 I | rafthttp: stopping peer a3da7b6cbdc85c13...
2016-07-08 00:27:52.581562 I | rafthttp: closed the TCP streaming connection with peer a3da7b6cbdc85c13 (stream MsgApp v2 writer)
2016-07-08 00:27:52.581720 I | rafthttp: stopped streaming with peer a3da7b6cbdc85c13 (writer)
2016-07-08 00:27:52.584025 I | rafthttp: closed the TCP streaming connection with peer a3da7b6cbdc85c13 (stream Message writer)
2016-07-08 00:27:52.584178 I | rafthttp: stopped streaming with peer a3da7b6cbdc85c13 (writer)
2016-07-08 00:27:52.584637 I | rafthttp: stopped HTTP pipelining with peer a3da7b6cbdc85c13
2016-07-08 00:27:52.585045 W | rafthttp: lost the TCP streaming connection with peer a3da7b6cbdc85c13 (stream MsgApp v2 reader)
2016-07-08 00:27:52.585198 I | rafthttp: stopped streaming with peer a3da7b6cbdc85c13 (stream MsgApp v2 reader)
2016-07-08 00:27:52.585576 W | rafthttp: lost the TCP streaming connection with peer a3da7b6cbdc85c13 (stream Message reader)
2016-07-08 00:27:52.585716 E | rafthttp: failed to read a3da7b6cbdc85c13 on stream Message (net/http: request canceled)
2016-07-08 00:27:52.585809 I | rafthttp: peer a3da7b6cbdc85c13 became inactive
2016-07-08 00:27:52.585926 I | rafthttp: stopped streaming with peer a3da7b6cbdc85c13 (stream Message reader)
2016-07-08 00:27:52.586074 I | rafthttp: stopped peer a3da7b6cbdc85c13
2016-07-08 00:27:52.586182 I | rafthttp: stopping peer d6ac5313ae37eb88...
2016-07-08 00:27:52.587816 I | rafthttp: closed the TCP streaming connection with peer d6ac5313ae37eb88 (stream MsgApp v2 writer)
2016-07-08 00:27:52.587941 I | rafthttp: stopped streaming with peer d6ac5313ae37eb88 (writer)
2016-07-08 00:27:52.589749 I | rafthttp: closed the TCP streaming connection with peer d6ac5313ae37eb88 (stream Message writer)
2016-07-08 00:27:52.589899 I | rafthttp: stopped streaming with peer d6ac5313ae37eb88 (writer)
2016-07-08 00:27:52.590262 I | rafthttp: stopped HTTP pipelining with peer d6ac5313ae37eb88
2016-07-08 00:27:52.590645 W | rafthttp: lost the TCP streaming connection with peer d6ac5313ae37eb88 (stream MsgApp v2 reader)
2016-07-08 00:27:52.590785 E | rafthttp: failed to read d6ac5313ae37eb88 on stream MsgApp v2 (net/http: request canceled)
2016-07-08 00:27:52.590875 I | rafthttp: peer d6ac5313ae37eb88 became inactive
2016-07-08 00:27:52.591010 I | rafthttp: stopped streaming with peer d6ac5313ae37eb88 (stream MsgApp v2 reader)
2016-07-08 00:27:52.591846 W | rafthttp: lost the TCP streaming connection with peer d6ac5313ae37eb88 (stream Message reader)
2016-07-08 00:27:52.592101 I | rafthttp: stopped streaming with peer d6ac5313ae37eb88 (stream Message reader)
2016-07-08 00:27:52.592302 I | v3rpc/grpc: transport: http2Client.notifyError got notified that the client transport was broken EOF.
2016-07-08 00:27:52.593249 I | rafthttp: stopped peer d6ac5313ae37eb88
2016-07-08 00:27:52.613271 W | etcdserver: failed to send out heartbeat on time (exceeded the 10ms timeout for 301.57µs)
2016-07-08 00:27:52.613463 W | etcdserver: server is likely overloaded
2016-07-08 00:27:52.613676 W | etcdserver: failed to send out heartbeat on time (exceeded the 10ms timeout for 719.54µs)
2016-07-08 00:27:52.613821 W | etcdserver: server is likely overloaded
2016-07-08 00:27:52.619323 I | integration: stopped node2587048822844350318 (unix://localhost:node2587048822844350318.sock.bridge)
2016-07-08 00:27:52.622414 W | rafthttp: lost the TCP streaming connection with peer 3299bc18a0c7b5b0 (stream Message reader)
2016-07-08 00:27:52.623244 W | rafthttp: lost the TCP streaming connection with peer 3299bc18a0c7b5b0 (stream MsgApp v2 reader)
2016-07-08 00:27:52.624723 W | rafthttp: lost the TCP streaming connection with peer 3299bc18a0c7b5b0 (stream Message reader)
2016-07-08 00:27:52.624951 W | rafthttp: lost the TCP streaming connection with peer 3299bc18a0c7b5b0 (stream MsgApp v2 reader)
2016-07-08 00:27:52.625301 E | rafthttp: failed to write 3299bc18a0c7b5b0 on stream Message (write unix 127.0.0.1:21123.29554.sock->@: write: broken pipe)
2016-07-08 00:27:52.625500 I | rafthttp: peer 3299bc18a0c7b5b0 became inactive
2016-07-08 00:27:52.627516 W | rafthttp: lost the TCP streaming connection with peer 3299bc18a0c7b5b0 (stream Message writer)
2016-07-08 00:27:52.724092 E | rafthttp: failed to dial 3299bc18a0c7b5b0 on stream Message (dial unix 127.0.0.1:21119.29554.sock: connect: no such file or directory)
2016-07-08 00:27:52.724259 I | rafthttp: peer 3299bc18a0c7b5b0 became inactive
2016-07-08 00:27:53.149713 W | rafthttp: lost the TCP streaming connection with peer 3299bc18a0c7b5b0 (stream MsgApp v2 writer)
2016-07-08 00:27:53.610062 I | v3rpc/grpc: grpc: addrConn.resetTransport failed to create client transport: connection error: desc = "transport: dial unix localhost:node2587048822844350318.sock.bridge: connect: no such file or directory"; Reconnecting to {"localhost:node2587048822844350318.sock.bridge" <nil>}
2016-07-08 00:27:53.636582 I | integration: restarting node2587048822844350318 (unix://localhost:node2587048822844350318.sock.bridge)
2016-07-08 00:27:53.637342 I | integration: launching node2587048822844350318 (unix://localhost:node2587048822844350318.sock.bridge)
2016-07-08 00:27:53.650952 I | etcdserver: name = node2587048822844350318
2016-07-08 00:27:53.651116 I | etcdserver: data dir = /tmp/etcd192011828
2016-07-08 00:27:53.651249 I | etcdserver: member dir = /tmp/etcd192011828/member
2016-07-08 00:27:53.651347 I | etcdserver: heartbeat = 10ms
2016-07-08 00:27:53.651457 I | etcdserver: election = 100ms
2016-07-08 00:27:53.651551 I | etcdserver: snapshot count = 0
2016-07-08 00:27:53.651688 I | etcdserver: advertise client URLs = unix://127.0.0.1:21120.29554.sock
2016-07-08 00:27:53.654587 I | etcdserver: restarting member 3299bc18a0c7b5b0 in cluster 14b0d63db98d5990 at commit index 9
2016-07-08 00:27:53.654947 I | raft: 3299bc18a0c7b5b0 became follower at term 3
2016-07-08 00:27:53.655134 I | raft: newRaft 3299bc18a0c7b5b0 [peers: [], term: 3, commit: 9, applied: 0, lastindex: 9, lastterm: 3]
2016-07-08 00:27:53.659348 I | etcdserver: set snapshot count to default 10000
2016-07-08 00:27:53.659506 I | etcdserver: starting server... [version: 3.0.0+git, cluster version: to_be_decided]
2016-07-08 00:27:53.680336 W | etcdserver: failed to send out heartbeat on time (exceeded the 10ms timeout for 33.120689ms)
2016-07-08 00:27:53.680490 W | etcdserver: server is likely overloaded
2016-07-08 00:27:53.680587 W | etcdserver: failed to send out heartbeat on time (exceeded the 10ms timeout for 33.405492ms)
2016-07-08 00:27:53.680650 W | etcdserver: server is likely overloaded
2016-07-08 00:27:53.685915 I | integration: launched node2587048822844350318 (unix://localhost:node2587048822844350318.sock.bridge)
2016-07-08 00:27:53.686032 I | integration: restarted node2587048822844350318 (unix://localhost:node2587048822844350318.sock.bridge)
2016-07-08 00:27:53.689235 I | raft: 3299bc18a0c7b5b0 is starting a new election at term 3
2016-07-08 00:27:53.689393 I | raft: 3299bc18a0c7b5b0 became candidate at term 4
2016-07-08 00:27:53.689496 I | raft: 3299bc18a0c7b5b0 received vote from 3299bc18a0c7b5b0 at term 4
2016-07-08 00:27:53.689627 I | raft: 3299bc18a0c7b5b0 became leader at term 4
2016-07-08 00:27:53.689734 I | raft: raft.node: 3299bc18a0c7b5b0 elected leader 3299bc18a0c7b5b0 at term 4
2016-07-08 00:27:53.690770 I | membership: added member 3299bc18a0c7b5b0 [unix://127.0.0.1:21119.29554.sock] to cluster 14b0d63db98d5990
2016-07-08 00:27:53.691840 I | membership: added member a3da7b6cbdc85c13 [unix://127.0.0.1:21121.29554.sock] to cluster 14b0d63db98d5990
2016-07-08 00:27:53.691991 I | rafthttp: starting peer a3da7b6cbdc85c13...
2016-07-08 00:27:53.693007 I | rafthttp: started HTTP pipelining with peer a3da7b6cbdc85c13
2016-07-08 00:27:53.699264 I | rafthttp: started peer a3da7b6cbdc85c13
2016-07-08 00:27:53.699735 I | rafthttp: added peer a3da7b6cbdc85c13
2016-07-08 00:27:53.700753 I | rafthttp: started streaming with peer a3da7b6cbdc85c13 (writer)
2016-07-08 00:27:53.700942 I | rafthttp: started streaming with peer a3da7b6cbdc85c13 (writer)
2016-07-08 00:27:53.701164 I | rafthttp: started streaming with peer a3da7b6cbdc85c13 (stream MsgApp v2 reader)
2016-07-08 00:27:53.701791 I | rafthttp: started streaming with peer a3da7b6cbdc85c13 (stream Message reader)
2016-07-08 00:27:53.702959 W | etcdserver: failed to send out heartbeat on time (exceeded the 10ms timeout for 2.624055ms)
2016-07-08 00:27:53.703070 W | etcdserver: server is likely overloaded
2016-07-08 00:27:53.703172 W | etcdserver: failed to send out heartbeat on time (exceeded the 10ms timeout for 2.583525ms)
2016-07-08 00:27:53.703253 W | etcdserver: server is likely overloaded
2016-07-08 00:27:53.704586 I | membership: added member d6ac5313ae37eb88 [unix://127.0.0.1:21123.29554.sock] to cluster 14b0d63db98d5990
2016-07-08 00:27:53.704810 I | rafthttp: starting peer d6ac5313ae37eb88...
2016-07-08 00:27:53.705824 I | rafthttp: started HTTP pipelining with peer d6ac5313ae37eb88
2016-07-08 00:27:53.712404 I | rafthttp: started peer d6ac5313ae37eb88
2016-07-08 00:27:53.712819 I | rafthttp: added peer d6ac5313ae37eb88
2016-07-08 00:27:53.714458 N | membership: set the initial cluster version to 3.0
2016-07-08 00:27:53.715476 I | etcdserver: published {Name:node2587048822844350318 ClientURLs:[unix://127.0.0.1:21120.29554.sock]} to cluster 14b0d63db98d5990
2016-07-08 00:27:53.717198 I | rafthttp: started streaming with peer d6ac5313ae37eb88 (writer)
2016-07-08 00:27:53.717360 I | rafthttp: started streaming with peer d6ac5313ae37eb88 (writer)
2016-07-08 00:27:53.717543 I | rafthttp: started streaming with peer d6ac5313ae37eb88 (stream MsgApp v2 reader)
2016-07-08 00:27:53.718532 I | rafthttp: started streaming with peer d6ac5313ae37eb88 (stream Message reader)
2016-07-08 00:27:53.736921 W | rafthttp: closed an existing TCP streaming connection with peer 3299bc18a0c7b5b0 (stream Message writer)
2016-07-08 00:27:53.737107 I | rafthttp: peer 3299bc18a0c7b5b0 became active
2016-07-08 00:27:53.737246 I | rafthttp: established a TCP streaming connection with peer 3299bc18a0c7b5b0 (stream Message writer)
2016-07-08 00:27:53.741667 W | rafthttp: closed an existing TCP streaming connection with peer 3299bc18a0c7b5b0 (stream MsgApp v2 writer)
2016-07-08 00:27:53.779063 I | rafthttp: peer 3299bc18a0c7b5b0 became active
2016-07-08 00:27:53.779153 I | rafthttp: established a TCP streaming connection with peer 3299bc18a0c7b5b0 (stream Message writer)
2016-07-08 00:27:53.781660 I | rafthttp: established a TCP streaming connection with peer 3299bc18a0c7b5b0 (stream MsgApp v2 writer)
2016-07-08 00:27:53.782302 I | rafthttp: peer d6ac5313ae37eb88 became active
2016-07-08 00:27:53.782351 I | rafthttp: established a TCP streaming connection with peer d6ac5313ae37eb88 (stream MsgApp v2 writer)
2016-07-08 00:27:53.782536 I | rafthttp: peer a3da7b6cbdc85c13 became active
2016-07-08 00:27:53.782578 I | rafthttp: established a TCP streaming connection with peer a3da7b6cbdc85c13 (stream Message writer)
2016-07-08 00:27:53.783199 I | rafthttp: established a TCP streaming connection with peer a3da7b6cbdc85c13 (stream MsgApp v2 writer)
2016-07-08 00:27:53.783251 I | rafthttp: established a TCP streaming connection with peer d6ac5313ae37eb88 (stream Message writer)
2016-07-08 00:27:53.783476 W | etcdserver: failed to send out heartbeat on time (exceeded the 10ms timeout for 60.545522ms)
2016-07-08 00:27:53.783508 W | etcdserver: server is likely overloaded
2016-07-08 00:27:53.783544 W | etcdserver: failed to send out heartbeat on time (exceeded the 10ms timeout for 60.387735ms)
2016-07-08 00:27:53.783567 W | etcdserver: server is likely overloaded
2016-07-08 00:27:53.784336 I | rafthttp: established a TCP streaming connection with peer a3da7b6cbdc85c13 (stream MsgApp v2 reader)
2016-07-08 00:27:53.785094 I | rafthttp: established a TCP streaming connection with peer a3da7b6cbdc85c13 (stream Message reader)
2016-07-08 00:27:53.785424 I | rafthttp: established a TCP streaming connection with peer 3299bc18a0c7b5b0 (stream MsgApp v2 reader)
2016-07-08 00:27:53.786133 I | rafthttp: established a TCP streaming connection with peer 3299bc18a0c7b5b0 (stream Message reader)
2016-07-08 00:27:53.786913 I | rafthttp: established a TCP streaming connection with peer 3299bc18a0c7b5b0 (stream MsgApp v2 writer)
2016-07-08 00:27:53.787469 I | rafthttp: established a TCP streaming connection with peer d6ac5313ae37eb88 (stream MsgApp v2 reader)
2016-07-08 00:27:53.789840 I | rafthttp: established a TCP streaming connection with peer 3299bc18a0c7b5b0 (stream Message reader)
2016-07-08 00:27:53.790114 I | rafthttp: established a TCP streaming connection with peer 3299bc18a0c7b5b0 (stream MsgApp v2 reader)
2016-07-08 00:27:53.790861 I | rafthttp: established a TCP streaming connection with peer d6ac5313ae37eb88 (stream Message reader)
2016-07-08 00:27:53.791628 I | raft: d6ac5313ae37eb88 [term: 3] received a MsgAppResp message with higher term from 3299bc18a0c7b5b0 [term: 4]
2016-07-08 00:27:53.791722 I | raft: d6ac5313ae37eb88 became follower at term 4
2016-07-08 00:27:53.791783 I | raft: raft.node: d6ac5313ae37eb88 changed leader from d6ac5313ae37eb88 to 3299bc18a0c7b5b0 at term 4
2016-07-08 00:27:53.793336 I | raft: d6ac5313ae37eb88 [term: 4] ignored a MsgAppResp message with lower term from a3da7b6cbdc85c13 [term: 3]
2016-07-08 00:27:53.793403 I | raft: d6ac5313ae37eb88 [term: 4] ignored a MsgHeartbeatResp message with lower term from a3da7b6cbdc85c13 [term: 3]
2016-07-08 00:27:53.793540 I | raft: d6ac5313ae37eb88 [term: 4] ignored a MsgHeartbeatResp message with lower term from a3da7b6cbdc85c13 [term: 3]
2016-07-08 00:27:53.795973 I | raft: a3da7b6cbdc85c13 [term: 3] received a MsgHeartbeat message with higher term from 3299bc18a0c7b5b0 [term: 4]
2016-07-08 00:27:53.796050 I | raft: a3da7b6cbdc85c13 became follower at term 4
2016-07-08 00:27:53.796097 I | raft: raft.node: a3da7b6cbdc85c13 changed leader from d6ac5313ae37eb88 to 3299bc18a0c7b5b0 at term 4
2016-07-08 00:27:53.798365 I | raft: d6ac5313ae37eb88 [term: 4] ignored a MsgHeartbeatResp message with lower term from a3da7b6cbdc85c13 [term: 3]
2016-07-08 00:27:53.798432 I | raft: d6ac5313ae37eb88 [term: 4] ignored a MsgHeartbeatResp message with lower term from a3da7b6cbdc85c13 [term: 3]
2016-07-08 00:27:55.094960 I | v3rpc/grpc: grpc: addrConn.resetTransport failed to create client transport: connection error: desc = "transport: dial unix localhost:node2587048822844350318.sock.bridge: connect: no such file or directory"; Reconnecting to {"localhost:node2587048822844350318.sock.bridge" <nil>}
2016-07-08 00:27:55.608327 I | integration: terminating node2587048822844350318 (unix://localhost:node2587048822844350318.sock.bridge)
2016-07-08 00:27:55.614761 I | rafthttp: stopping peer a3da7b6cbdc85c13...
2016-07-08 00:27:55.616894 I | rafthttp: closed the TCP streaming connection with peer a3da7b6cbdc85c13 (stream MsgApp v2 writer)
2016-07-08 00:27:55.617060 I | rafthttp: stopped streaming with peer a3da7b6cbdc85c13 (writer)
2016-07-08 00:27:55.619067 I | rafthttp: closed the TCP streaming connection with peer a3da7b6cbdc85c13 (stream Message writer)
2016-07-08 00:27:55.619222 I | rafthttp: stopped streaming with peer a3da7b6cbdc85c13 (writer)
2016-07-08 00:27:55.619811 I | rafthttp: stopped HTTP pipelining with peer a3da7b6cbdc85c13
2016-07-08 00:27:55.620301 W | rafthttp: lost the TCP streaming connection with peer a3da7b6cbdc85c13 (stream MsgApp v2 reader)
2016-07-08 00:27:55.620515 E | rafthttp: failed to read a3da7b6cbdc85c13 on stream MsgApp v2 (net/http: request canceled)
2016-07-08 00:27:55.620632 I | rafthttp: peer a3da7b6cbdc85c13 became inactive
2016-07-08 00:27:55.620799 I | rafthttp: stopped streaming with peer a3da7b6cbdc85c13 (stream MsgApp v2 reader)
2016-07-08 00:27:55.621284 W | rafthttp: lost the TCP streaming connection with peer a3da7b6cbdc85c13 (stream Message reader)
2016-07-08 00:27:55.621470 I | rafthttp: stopped streaming with peer a3da7b6cbdc85c13 (stream Message reader)
2016-07-08 00:27:55.621631 I | rafthttp: stopped peer a3da7b6cbdc85c13
2016-07-08 00:27:55.621757 I | rafthttp: stopping peer d6ac5313ae37eb88...
2016-07-08 00:27:55.623800 I | rafthttp: closed the TCP streaming connection with peer d6ac5313ae37eb88 (stream MsgApp v2 writer)
2016-07-08 00:27:55.623948 I | rafthttp: stopped streaming with peer d6ac5313ae37eb88 (writer)
2016-07-08 00:27:55.626154 I | rafthttp: closed the TCP streaming connection with peer d6ac5313ae37eb88 (stream Message writer)
2016-07-08 00:27:55.626318 I | rafthttp: stopped streaming with peer d6ac5313ae37eb88 (writer)
2016-07-08 00:27:55.626772 I | rafthttp: stopped HTTP pipelining with peer d6ac5313ae37eb88
2016-07-08 00:27:55.627335 W | rafthttp: lost the TCP streaming connection with peer d6ac5313ae37eb88 (stream MsgApp v2 reader)
2016-07-08 00:27:55.627531 E | rafthttp: failed to read d6ac5313ae37eb88 on stream MsgApp v2 (net/http: request canceled)
2016-07-08 00:27:55.627650 I | rafthttp: peer d6ac5313ae37eb88 became inactive
2016-07-08 00:27:55.627816 I | rafthttp: stopped streaming with peer d6ac5313ae37eb88 (stream MsgApp v2 reader)
2016-07-08 00:27:55.628277 W | rafthttp: lost the TCP streaming connection with peer d6ac5313ae37eb88 (stream Message reader)
2016-07-08 00:27:55.628517 I | rafthttp: stopped streaming with peer d6ac5313ae37eb88 (stream Message reader)
2016-07-08 00:27:55.628652 I | rafthttp: stopped peer d6ac5313ae37eb88
2016-07-08 00:27:55.663485 I | integration: terminated node2587048822844350318 (unix://localhost:node2587048822844350318.sock.bridge)
2016-07-08 00:27:55.663618 I | integration: terminating node5021162303408691792 (unix://localhost:node5021162303408691792.sock.bridge)
2016-07-08 00:27:55.666040 W | rafthttp: lost the TCP streaming connection with peer 3299bc18a0c7b5b0 (stream MsgApp v2 reader)
2016-07-08 00:27:55.666735 W | rafthttp: lost the TCP streaming connection with peer 3299bc18a0c7b5b0 (stream MsgApp v2 reader)
2016-07-08 00:27:55.667163 W | rafthttp: lost the TCP streaming connection with peer 3299bc18a0c7b5b0 (stream Message reader)
2016-07-08 00:27:55.668103 W | rafthttp: lost the TCP streaming connection with peer 3299bc18a0c7b5b0 (stream Message reader)
2016-07-08 00:27:55.668539 I | rafthttp: stopping peer 3299bc18a0c7b5b0...
2016-07-08 00:27:55.669986 I | rafthttp: closed the TCP streaming connection with peer 3299bc18a0c7b5b0 (stream MsgApp v2 writer)
2016-07-08 00:27:55.670121 I | rafthttp: stopped streaming with peer 3299bc18a0c7b5b0 (writer)
2016-07-08 00:27:55.671624 I | rafthttp: closed the TCP streaming connection with peer 3299bc18a0c7b5b0 (stream Message writer)
2016-07-08 00:27:55.671731 I | rafthttp: stopped streaming with peer 3299bc18a0c7b5b0 (writer)
2016-07-08 00:27:55.672951 I | rafthttp: stopped HTTP pipelining with peer 3299bc18a0c7b5b0
2016-07-08 00:27:55.673228 I | rafthttp: stopped streaming with peer 3299bc18a0c7b5b0 (stream MsgApp v2 reader)
2016-07-08 00:27:55.673500 I | rafthttp: stopped streaming with peer 3299bc18a0c7b5b0 (stream Message reader)
2016-07-08 00:27:55.673669 I | rafthttp: stopped peer 3299bc18a0c7b5b0
2016-07-08 00:27:55.673801 I | rafthttp: stopping peer d6ac5313ae37eb88...
2016-07-08 00:27:55.675835 I | rafthttp: closed the TCP streaming connection with peer d6ac5313ae37eb88 (stream MsgApp v2 writer)
2016-07-08 00:27:55.675996 I | rafthttp: stopped streaming with peer d6ac5313ae37eb88 (writer)
2016-07-08 00:27:55.678230 I | rafthttp: closed the TCP streaming connection with peer d6ac5313ae37eb88 (stream Message writer)
2016-07-08 00:27:55.678392 I | rafthttp: stopped streaming with peer d6ac5313ae37eb88 (writer)
2016-07-08 00:27:55.678907 I | rafthttp: stopped HTTP pipelining with peer d6ac5313ae37eb88
2016-07-08 00:27:55.679406 W | rafthttp: lost the TCP streaming connection with peer d6ac5313ae37eb88 (stream MsgApp v2 reader)
2016-07-08 00:27:55.679610 I | rafthttp: stopped streaming with peer d6ac5313ae37eb88 (stream MsgApp v2 reader)
2016-07-08 00:27:55.680171 W | rafthttp: lost the TCP streaming connection with peer d6ac5313ae37eb88 (stream Message reader)
2016-07-08 00:27:55.680346 E | rafthttp: failed to read d6ac5313ae37eb88 on stream Message (net/http: request canceled)
2016-07-08 00:27:55.680481 I | rafthttp: peer d6ac5313ae37eb88 became inactive
2016-07-08 00:27:55.680629 I | rafthttp: stopped streaming with peer d6ac5313ae37eb88 (stream Message reader)
2016-07-08 00:27:55.680820 I | rafthttp: stopped peer d6ac5313ae37eb88
2016-07-08 00:27:55.711160 I | integration: terminated node5021162303408691792 (unix://localhost:node5021162303408691792.sock.bridge)
2016-07-08 00:27:55.711304 I | integration: terminating node6227899913783768578 (unix://localhost:node6227899913783768578.sock.bridge)
2016-07-08 00:27:55.712920 I | rafthttp: stopping peer 3299bc18a0c7b5b0...
2016-07-08 00:27:55.714392 I | rafthttp: closed the TCP streaming connection with peer 3299bc18a0c7b5b0 (stream MsgApp v2 writer)
2016-07-08 00:27:55.714505 I | rafthttp: stopped streaming with peer 3299bc18a0c7b5b0 (writer)
2016-07-08 00:27:55.716065 I | rafthttp: closed the TCP streaming connection with peer 3299bc18a0c7b5b0 (stream Message writer)
2016-07-08 00:27:55.716178 I | rafthttp: stopped streaming with peer 3299bc18a0c7b5b0 (writer)
2016-07-08 00:27:55.716664 I | rafthttp: stopped HTTP pipelining with peer 3299bc18a0c7b5b0
2016-07-08 00:27:55.716830 I | rafthttp: stopped streaming with peer 3299bc18a0c7b5b0 (stream MsgApp v2 reader)
2016-07-08 00:27:55.716975 I | rafthttp: stopped streaming with peer 3299bc18a0c7b5b0 (stream Message reader)
2016-07-08 00:27:55.717090 I | rafthttp: stopped peer 3299bc18a0c7b5b0
2016-07-08 00:27:55.717176 I | rafthttp: stopping peer a3da7b6cbdc85c13...
2016-07-08 00:27:55.718595 I | rafthttp: closed the TCP streaming connection with peer a3da7b6cbdc85c13 (stream MsgApp v2 writer)
2016-07-08 00:27:55.718699 I | rafthttp: stopped streaming with peer a3da7b6cbdc85c13 (writer)
2016-07-08 00:27:55.720066 I | rafthttp: closed the TCP streaming connection with peer a3da7b6cbdc85c13 (stream Message writer)
2016-07-08 00:27:55.720166 I | rafthttp: stopped streaming with peer a3da7b6cbdc85c13 (writer)
2016-07-08 00:27:55.720739 I | rafthttp: stopped HTTP pipelining with peer a3da7b6cbdc85c13
2016-07-08 00:27:55.721462 W | rafthttp: lost the TCP streaming connection with peer a3da7b6cbdc85c13 (stream MsgApp v2 reader)
2016-07-08 00:27:55.721602 I | rafthttp: stopped streaming with peer a3da7b6cbdc85c13 (stream MsgApp v2 reader)
2016-07-08 00:27:55.722882 W | rafthttp: lost the TCP streaming connection with peer a3da7b6cbdc85c13 (stream Message reader)
2016-07-08 00:27:55.723000 I | rafthttp: stopped streaming with peer a3da7b6cbdc85c13 (stream Message reader)
2016-07-08 00:27:55.723102 I | rafthttp: stopped peer a3da7b6cbdc85c13
2016-07-08 00:27:55.754305 I | integration: terminated node6227899913783768578 (unix://localhost:node6227899913783768578.sock.bridge)
--- FAIL: TestLeaseKeepAliveCloseAfterDisconnectRevoke (3.74s)
	lease_test.go:380: unexpected keepalive cluster_id:1490927037254031760 member_id:3646152187295806896 revision:1 raft_term:4 
```

--- a/clientv3/integration/lease_test.go
+++ b/clientv3/integration/lease_test.go
@@ -359,7 +359,8 @@ func TestLeaseKeepAliveCloseAfterDisconnectRevoke(t *testing.T) {
 	if kerr != nil {
 		t.Fatal(kerr)
 	}
-	if kresp := <-rc; kresp.ID != resp.ID {
+	kresp := <-rc
+	if kresp.ID != resp.ID {
 		t.Fatalf("ID = %x, want %x", kresp.ID, resp.ID)
 	}
 
@@ -374,13 +375,14 @@ func TestLeaseKeepAliveCloseAfterDisconnectRevoke(t *testing.T) {
 
 	clus.Members[0].Restart(t)
 
-	select {
-	case ka, ok := <-rc:
-		if ok {
-			t.Fatalf("unexpected keepalive %v", ka)
+	// some keep-alives may still be buffered; drain until close
+	timer := time.After(time.Duration(kresp.TTL) * time.Second)
+	for kresp != nil {
+		select {
+		case kresp = <-rc:
+		case <-timer:
+			t.Fatalf("keepalive channel did not close")
 		}
-	case <-time.After(5 * time.Second):
-		t.Fatalf("keepalive channel did not close")
 	}
 }
 

